File: projects/GPL-FH-Java/features/DirectedWithEdges/GPL/Vertex.java
package GPL;

import java.util.LinkedList;
import java.util.Iterator;

// of graph

  // ***********************************************************************

public class Vertex {

    // dja: changed neighbors and name to public
    public LinkedList neighbors;

    public String name;

    public String getName() { return name; }

    public Vertex() {
        VertexConstructor();
    }

    public void VertexConstructor() {
        name      = null;
        neighbors = new LinkedList();
    }

    public  Vertex assignName( String name ) {
        this.name = name;
        return ( Vertex ) this;
    }

    public void addNeighbor( Neighbor n ) {
        neighbors.add( n );
    }

    public VertexIter getNeighbors() {
        return new VertexIter() {
                private Iterator iter = neighbors.iterator();
                public Vertex next() { return ((Neighbor)iter.next()).end; }
                public boolean hasNext() { return iter.hasNext(); }
            };
    }

    public EdgeIter getEdges()
    {
        return new EdgeIter()
            {
                private Iterator iter = neighbors.iterator();
                /* dja: changed to fix compile error */
//                public EdgeIfc next() { return ((EdgeIfc)  iter.next()).edge; }
                public EdgeIfc next( ) 
                { 
                  return ( ( EdgeIfc ) ( ( Neighbor ) iter.next( ) ).edge ); 
                }
                public boolean hasNext() { return iter.hasNext(); }
            };
    }

    public void display() {
        System.out.print( " Node " + getName() + " connected to: " );

        for(VertexIter vxiter = getNeighbors(); vxiter.hasNext(); )
         {
            Vertex v = vxiter.next();
            System.out.print( v.getName() + ", " );
        }

        System.out.println();
    }
}

File: plugins/de.ovgu.featureide.examples/featureide_examples/GPL-FH-CSharp/features/DirectedWithEdges/Vertex.cs
//layer Directed;

using System.Collections;
using System.Collections.Generic;
//import java.util.Iterator;

// of graph

  // ***********************************************************************

public class Vertex {

    // dja: changed neighbors and name to public
    public List<Neighbor> neighbors;

    public string name;

    public string GetName() { return name; }

    public Vertex() {
        VertexConstructor();
    }

    public void VertexConstructor() {
        name      = null;
        neighbors = new List<Neighbor>();
    }

    public  Vertex assignName( string name ) {
        this.name = name;
        return ( Vertex ) this;
    }

    public void AddNeighbor( Neighbor n ) {
        neighbors.Add( n );
    }

    public Iterator<Vertex> GetNeighbors() {
        return new Iterator<Vertex, Neighbor>(neighbors.GetEnumerator(), TransformNeighborToEnd);
    }
    public static Vertex TransformNeighborToEnd(Neighbor val) {
    	// should be a Lambda Method in C# 3.0
        return val.end;
    }
    public Iterator<EdgeIfc> GetEdges() {
        return new Iterator<EdgeIfc, Neighbor>(neighbors.GetEnumerator(), TransformNeighborToEdge);
    }
    public static EdgeIfc TransformNeighborToEdge(Neighbor val) {
    	// should be a Lambda Method in C# 3.0
        return val.edge;
    }

    public void display() {
        System.Console.Out.Write( " Node " + GetName() + " connected to: " );

        for(Iterator<Vertex> vxiter = GetNeighbors(); vxiter.hasNext(); )
         {
            Vertex v = vxiter.next();
            System.Console.Out.Write( v.GetName() + ", " );
        }

        System.Console.Out.WriteLine();
    }
}
